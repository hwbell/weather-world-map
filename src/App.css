
* {
  font-family: Manjari;
  color: whitesmoke;
}

.fa .fas .fab {
  font-size: 55px;
}
.App {
  text-align: center;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
