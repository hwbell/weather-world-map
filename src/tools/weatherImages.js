// need function to return an icon specifier, given the code from darksky
// icon options from the dark sky api:
// clear-day, clear-night, partly-cloudy-day, partly-cloudy-night, rain, snow, sleet, wind, fog, cloudy
// return the class of the font-awesome icon to be used

const weatherImagesList = {
  'clear-day': {
    icon: 'fas fa-sun',
  },
  'clear-night': {
    icon: 'fas fa-star-and-crescent'
  },
  'partly-cloudy-day': {
    icon: 'fas fa-cloud-sun'
  },
  'partly-cloudy-night': {
    icon: 'fas fa-cloud-moon'
  },
  'cloudy': {
    icon: 'fas fa-cloud'
  },
  'fog': {
    icon: 'fas fa-smog'
  },
  'rain': {
    icon: 'fas fa-cloud-rain'
  },
  'wind': {
    icon: 'fas fa-wind'
  },
  'sleet': {
    icon: 'fas fa-cloud-showers-heavy'
  },
  'snow': {
    icon: 'fas fa-snowflake'
  },

}

// this just indexes the icon list by id, returning the correct class
const getWeatherIcon = (iconId) => {
  return weatherImagesList[iconId].icon;
}

module.exports = {
  weatherImagesList,
  getWeatherIcon,
}