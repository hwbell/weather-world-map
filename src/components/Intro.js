import React, { Component } from 'react';

// style
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// animation with pose
import posed from 'react-pose';

// use react-pose for fading in
const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

const P = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

class Intro extends Component {

  constructor(props) {
    super(props);
    this.state = {
      isVisible: false
    }
  }

  componentDidMount() {
    setTimeout(() => {
      this.setState({ isVisible: true });
    }, 800);
  }

  render() {
    const isVisible = this.state.isVisible;

    let lat = Math.round(this.props.coords.lat * 100) / 100;
    let lng = Math.round(this.props.coords.lng * 100) / 100;
    
    return (
      <Div pose={isVisible ? 'visible' : 'hidden'} style={styles.container}>

        <P style={styles.title}>take a look around</P>
        <P style={styles.subtitle}>click anywhere you'd like to see the weather</P>

        <div className="">
          <P style={styles.coords}>{`latitude: ${lat}`}</P>
          <P style={styles.coords}>{`longitude: ${lng}`}</P>
        </div>

      </Div>
    );
  }
}

const styles = {
  container: {
    zIndex: 1,
    position: 'absolute',
    top: '30px',
    left: '20px',
  },
  title: {
    marginLeft: '20px',
    color: 'black',
    fontWeight: 'bold',
    fontSize: 'calc(16px + 2vw)'
  },
  subtitle: {
    margin: '20px',
    color: 'black',
    fontWeight: 'bold',
    fontSize: 'calc(10px + 1vw)'
  },
  coords: {
    margin: '20px',
    color: 'black',
    fontWeight: 'bold',
    fontSize: 'calc(6px + 1vw)'
  },
}

export default Intro;
