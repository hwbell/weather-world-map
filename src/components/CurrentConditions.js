import React, { Component } from 'react';

// style
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// animation with pose
import posed from 'react-pose';

// tools
import { weatherImagesList, getWeatherIcon } from '../tools/weatherImages';

// use react-pose for fading in
const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

const P = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

class CurrentConditions extends Component {

  constructor(props) {
    super(props);
    this.state = {
      // 
    }
  }

  componentDidMount() {
    // 
  }

  render() {
    let weather = this.props.weather;

    return (
      <div className="row" style={styles.container}>

        <div className="col">
          <i style={styles.icon} className={getWeatherIcon(weather.icon) + ' col'}></i>
        </div>

        <div className="col">
            <p style={styles.temperature}>{weather.temperature}&deg;F</p>
        </div>

        <div className="col">
          <p style={styles.paragraph}>{weather.summary}</p>
          <p style={styles.paragraph}>{`${weather.humidity}% humidity`}</p>
        </div>

        <div className="col">
          <p style={styles.paragraph}>{`${weather.date}`}</p>
          <p style={styles.paragraph}>{`${weather.time}`}</p>
        </div>

      </div>
    );
  }
}

const styles = {
  container: {
    margin: '10',
    marginLeft: '10vw'
  },
  icon: {
    fontSize: '60px',
    padding: '10px'
  },
  tempHolder: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  temperature: {
    fontSize: '44px',
  },
  paragraph: {
    color: 'whitesmoke',
    fontSize: 'calc(10px + 1vw)',
    marginHorizontal: '20px',
    // paddingHorizontal: 12
  }
}

export default CurrentConditions;
