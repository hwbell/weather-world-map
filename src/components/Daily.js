import React, { Component } from 'react';

// styles
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button } from 'reactstrap';
import HorizontalScroller from 'react-horizontal-scroll-container';

// animation with pose
import posed from 'react-pose';

// tools
import { weatherImageList, getWeatherIcon } from '../tools/weatherImages';
var Rainbow = require('rainbowvis.js');

var ScrollArea = require('react-scrollbar');

// use react-pose for fading in
const Container = posed.div({
  enter: { staggerChildren: 50 },
  exit: { staggerChildren: 50, staggerDirection: -1 }
});

const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

class Hourly extends Component {

  constructor(props) {

    super(props);
    this.state = {
      //
    }
  }

  componentDidMount() {

  }

  renderWeatherInfo() {

    // initalize rainbowvis to color each group dynamically
    var myRainbow = new Rainbow();

    // make a spectrum based on the first temp in the list
    let temp = this.props.weatherList[0].temperatureHigh;
    // set range based on data
    myRainbow.setNumberRange(-15, 100);
    myRainbow.setSpectrum('#303F9F', '#FF5722');

    let hourlyContainers = this.props.weatherList.map((weather, i) => {

      let style = {
        minWidth: '160px',
        padding: 0,
        margin: 0,
        border: '1px solid rgba(245, 245, 245, 0.2)',
        backgroundColor: '#' + myRainbow.colourAt(weather.temperatureHigh),
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'center'
      }

      return (
        <Div key={i} className="" style={style}>

          {/* the timestamp */}
          <div className="text-center row" style={styles.date}>
            <p style={styles.text}>{weather.date}</p>
          </div>

          {/* the temp and icon */}
          <div className="row">
            <div className="col">
              <i style={styles.icon} className={getWeatherIcon(weather.icon)}></i>
            </div>
            <div className="col">
              <p style={styles.highTempText}>{weather.temperatureHigh} &deg;F</p>
              <p style={styles.lowTempText}>{weather.temperatureLow} &deg;F</p>
            </div>
          </div>

          <p style={styles.paragraph}>
            {weather.summary}
          </p>

        </Div>
      )
    });

    return hourlyContainers;
  }

  render() {
    // console.log(this.props.weatherList)

    return (
      <Container className="" style={styles.container}>

        {/* <p style={styles.title}>daily</p> */}

        <div className="" style={{borderRadius: '55px'}}>
          <HorizontalScroller className="horizontal-scroller" style={styles.scroller}>
            {this.props.weatherList &&
              this.renderWeatherInfo()
            }
          </HorizontalScroller>
        </div>


      </Container>
    );
  }
}

const styles = {
  container: {
    width: '95%',
    margin: 'auto auto',
  },
  scroller: {
    overflowX: 'hidden',
    // margin: '0px',
    // padding: '0px'
  },
  title: {
    margin: '5px',
    fontSize: '12px',
    alignSelf: 'flex-start'
  },
  icon: {
    fontSize: '45px',
    padding: 10,
    paddingTop: 5
  },
  highTempText: {
    fontSize: '20px'
  },
  lowTempText: {
    fontSize: '14px'
  },
  date: {
    marginTop: '10px'
  },
  text: {
    textAlign: 'center',
    minWidth: '50px',
    paddingHorizontal: 10,
    fontSize: '18px',
    fontWeight: 'bold'
  },
  paragraph: {
    textAlign: 'left',
    margin: '8px',
    marginTop: '20px',
    fontSize: '14px',
  }
}

export default Hourly;
