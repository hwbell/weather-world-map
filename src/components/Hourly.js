import React, { Component } from 'react';

// styles
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button } from 'reactstrap';
import HorizontalScroller from 'react-horizontal-scroll-container';

// animation with pose
import posed from 'react-pose';

// tools
import { weatherImageList, getWeatherIcon } from '../tools/weatherImages';
var Rainbow = require('rainbowvis.js');

// use react-pose for fading in
const Container = posed.div({
  enter: { staggerChildren: 50 },
  exit: { staggerChildren: 50, staggerDirection: -1 }
});

const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

class Hourly extends Component {

  constructor(props) {

    super(props);
    this.state = {
      //
    }
  }

  componentDidMount() {

  }

  renderWeatherInfo() {

    // initalize rainbowvis to color each group dynamically
    var myRainbow = new Rainbow();

    // make a spectrum based on the first temp in the list
    let temp = this.props.weatherList[0].temp;
    // set range based on data
    myRainbow.setNumberRange(0, 100);
    myRainbow.setSpectrum('#1976D2', '#FF5722');

    let hourlyContainers = this.props.weatherList.map((weather, i) => {

      let style = {
        minWidth: '140px',
        backgroundColor: '#' + myRainbow.colourAt(weather.temp)
      }

      return (
        <Div key={i} className="" style={style}>

          {/* the temp and icon */}
          <div className="">
            <i style={styles.icon} className={getWeatherIcon(weather.icon)}></i>
          </div>
          <div className="">
            <p style={styles.temp}>{weather.temp} &deg;F</p>
          </div>

          {/* the timestamp */}
          <p style={styles.text}>{weather.time}</p>
          <p style={styles.text}>{weather.date}</p>

        </Div>
      )
    });

    return hourlyContainers;
  }

  render() {
    // console.log(this.props.weatherList)

    return (
      <Container className="" style={styles.container}>

        <div className="row">
          <HorizontalScroller>
            {this.props.weatherList &&
              this.renderWeatherInfo()
            }
          </HorizontalScroller>
        </div>


      </Container>
    );
  }
}

const styles = {
  container: {
    padding: '0px',
    margin: '15px'
  },

  temp: {
    textAlign: 'center',
    margin: 5,
    fontSize: '24px'
  },
  icon: {
    fontSize: '40px',
    textAlign: 'center',
    padding: '15px',
    paddingTop: '10px'
  },
  text: {
    textAlign: 'center',
    paddingHorizontal: 10
  }

}

export default Hourly;
