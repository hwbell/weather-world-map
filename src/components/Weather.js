import React, { Component } from 'react';

// styles
import '../App.css';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button, Modal } from 'reactstrap';
import Hourly from './Hourly';
import Daily from './Daily';
import CurrentConditions from './CurrentConditions';

// animation with pose
import posed, { PoseGroup } from 'react-pose';

// tools
import { weatherImageList, getWeatherIcon } from '../tools/weatherImages';

// use react-pose for fading in
const Div = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

const P = posed.div({
  hidden: { opacity: 0 },
  visible: { opacity: 1 }
});

// need arrays of Months and Days to get names
const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];

// getCurrentData() gets condensed info for the CurrentConditions component, 
// from res.currently we'll pass in res.currently from the darksky res
const getCurrentData = (currentData) => {

  // Dark Sky API makes this pretty convenient

  // get the date from the provided unix time 
  const time = new Date(currentData.time * 1000);

  let year = time.getFullYear().toString();
  let date = time.getDate();
  let month = months[time.getMonth()];

  let hours = time.getHours() > 12 ? time.getHours() - 11 : time.getHours() + 1;
  let minutes = time.getMinutes() < 10 ? '0' + time.getMinutes.toString() : time.getMinutes();
  let notation = time.getHours() > 10 ? ' pm' : ' am';

  let { humidity, icon, summary, temperature, uvIndex, windSpeed } = currentData;

  return {
    humidity, icon, summary, uvIndex, windSpeed,
    temperature: Math.floor(temperature),
    time: `${hours}:00 ${notation}`,
    date: `${month} ${date}, ${year}`
  };
}

// makeHourlyList() gets condensed info hourly info from the main response's hourly data
// we'll pass in res.hourly from the darksky res
const makeHourlyList = (hourlyData) => {

  // Dark Sky API makes this pretty convenient

  let condensedList = []; // we'll make a simple array for the days forecasted
  hourlyData.forEach((hour) => {
    // get the date from the provided unix time 
    const time = new Date(hour.time * 1000);

    let year = time.getFullYear().toString().slice(0, 2);
    let date = time.getDate();
    let month = months[time.getMonth()];

    let hours = time.getHours() > 12 ? time.getHours() - 11 : time.getHours() + 1;
    let notation = time.getHours() > 10 ? ' pm' : ' am';

    condensedList.push({
      desc: hour.summary,
      icon: hour.icon,
      temp: Math.floor(hour.temperature),
      time: `${hours}:00 ${notation}`,
      date: `${month} ${date}`
    });

  });

  return condensedList;
}

// makeHourlyList() gets condensed info hourly info from the main response's hourly data
// we'll pass in res.daily from the darksky res
const makeDailyList = (dailyData) => {

  // Dark Sky API makes this pretty convenient

  let condensedList = []; // we'll make a simple array for the days forecasted
  dailyData.forEach((day) => {
    // get the date from the provided unix time 
    const time = new Date(day.time * 1000);

    // let year = time.getFullYear().toString();
    let date = time.getDate();
    let month = months[time.getMonth()];
    let weekday = days[time.getDay()];

    let {
      temperatureHigh,
      temperatureLow,
      summary,
      humidity,
      icon,
      windSpeed,
    } = day;

    condensedList.push({
      temperatureHigh: Math.floor(temperatureHigh),
      temperatureLow: Math.floor(temperatureLow),
      summary,
      humidity,
      icon,
      windSpeed,
      date: `${weekday}, ${month} ${date}` // need to correct for the api's month array
    });

  });

  return condensedList;
}



class Weather extends Component {

  constructor(props) {

    super(props);
    this.state = {
      showDaily: true,
      showHourly: false
    }
    this.handleSwitch = this.handleSwitch.bind(this);
  }

  componentDidMount() {

  }

  // for the toggle between hourly and daily
  handleSwitch() {
    this.setState({
      showDaily: !this.state.showDaily,
      showHourly: !this.state.showHourly
    })
  }

  render() {
    const isVisible = this.props.showWeather;
    const data = this.props.weatherData;

    // make condensed lists to pass to the CurrentConditions, Hourly, Daily components
    let hourlyWeather = makeHourlyList(data.hourly.data);

    let currentWeather = getCurrentData(data.currently);

    // only pass the first 7 days, this list is really long
    let dailyWeather = makeDailyList(data.daily.data).slice(0, 8);

    // change the summary based on hourly vs. daily
    let summary = this.state.showDaily ? data.daily.summary : data.hourly.summary;

    return (

      <Div pose={isVisible ? 'visible' : 'hidden'} style={styles.container}>

        <Button color='link' onClick={this.props.close} className="float-right">
          <i style={styles.icon} className="fas fa-times-circle"></i>
        </Button>

        <p style={styles.title}>{this.props.location}</p>


        {/* the current conditions */}
        {currentWeather &&
          <CurrentConditions
            weather={currentWeather} />}

        {/* the selestor for hourly / daily */}
        <Button color="link" style={styles.switchButton}
          onClick={this.handleSwitch}
        >
          switch to {this.state.showHourly ? 'daily' : 'hourly'}
        </Button>

        {/* animate the switch */}
        <PoseGroup>
          <Div key={'hourly'}>
            {/* the hourly weather tabs  */}
            {this.state.showHourly && <Hourly weatherList={hourlyWeather} />}
          </Div>

          <Div key={'daily'}>
            {/* the daily weather tabs */}
            {this.state.showDaily && <Daily weatherList={dailyWeather} />}
          </Div>
        </PoseGroup>

        {/* short summary */}
        <p style={styles.subtitle}>{summary}</p>

      </Div>
    );
  }
}

const styles = {
  container: {
    backgroundColor: 'rgba(0,0,0,0.75)',
    zIndex: 1,
    position: 'absolute',
    bottom: '10px',
    height: '550px',
    width: '100%',
  },
  icon: {
    fontSize: '24px'
  },
  title: {
    fontWeight: 'bold',
    color: 'whitesmoke',
    fontSize: 'calc(14px + 1vw)',
    marginLeft: '5vw',
    padding: '10px'
  },
  subtitle: {
    color: 'whitesmoke',
    fontSize: 'calc(10px + 1vw)',
    margin: '3vh',
    marginLeft: '10vw',
  },
  switchButton: {
    textDecoration: 'none',
    fontSize: '20px',
    fontWeight: 'bold',
    marginLeft: '9vw'
  }
}

export default Weather;
