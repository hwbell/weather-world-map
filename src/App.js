import React, { Component } from 'react';

// styles
import './App.css';
import 'bootstrap/dist/css/bootstrap.css';

// components
import Intro from './components/Intro';
import Weather from './components/Weather';

// tools
import mapboxgl from 'mapbox-gl';
const fetch = require('node-fetch');

class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      showWeather: false 
    }
    this.renderMap = this.renderMap.bind(this);
    this.fetchWeather = this.fetchWeather.bind(this);
    this.toggleShowWeather = this.toggleShowWeather.bind(this);
  }

  componentDidMount() {
    this.renderMap();
  }

  // this is called both rom within the Weather component, to close the 
  // weather window
  toggleShowWeather() {
    this.setState({
      showWeather: !this.state.showWeather
    })
  }

  // get the weather json to be passed to the Weather component
  fetchWeather() {
    const self = this;

    let lat = this.state.weatherCoords.lat;
    let lng = this.state.weatherCoords.lng;

    fetch(`https://hb-weather-server.herokuapp.com/weather/${lat}/${lng}`, {
      method: 'GET',
      headers: {
        "Content-type": "application/json"
      }
    })
      .then( res => res.json() )
      .then( res => {
        console.log(res);
        self.setState({
          weatherData: res,
          showWeather: true
        });
        
      })
  }

  // show public map
  renderMap() {
    const self = this;

    mapboxgl.accessToken = 'pk.eyJ1IjoiaHdiZWxsIiwiYSI6ImNqdXU4OXY2YjA4cWU0NGxsZzFlYWdobmwifQ.nnjWcSrzbW4o3oq-QYOXhg';
    var map = new mapboxgl.Map({
      container: 'map',
      style: 'mapbox://styles/mapbox/streets-v11'
    });

    map.on('mousemove', function (e) {

      // save the state of the mouse in lat, lng (coords) and  x,y (points) 
      self.setState({
        coords: e.lngLat,
        points: e.point
      });

    })

    // handle the user clicking the map
    map.on('click', function (e) {

      // update state 
      self.setState({
        weatherCoords: e.lngLat,
        points: e.point
      }, () => {
        self.fetchWeather();
      });
    });
  }

  render() {
    return (
      <div id='map' style={{ width: '1100px', height: '800px' }}>

        {/* shows after 2000ms */}
        { this.state.coords && <Intro />}

        {/* shows when the user clicks on the map */}
        { this.state.showWeather && 
        <Weather 
          weather={this.state.weatherData}
          coords={this.state.weatherCoords}
          showWeather={this.state.showWeather}
          toggle={this.toggleShowWeather}
        /> }

      </div>
    );
  }
}

export default App;
