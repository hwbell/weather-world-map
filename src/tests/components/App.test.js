import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import App from '../../App';
import CurrentConditions from '../../components/CurrentConditions';
import Daily from '../../components/Daily';
import Hourly from '../../components/Hourly';
import Intro from '../../components/Intro';
import Weather from '../../components/Weather';

import { getWeatherIcon } from '../../tools/weatherImages';

let mockWeatherObj = {
  currently: {
    data: []
  }, 
  hourly: {
    data: []
  }, 
  daily: {
    data: []
  }
}


// I got this mock documentation from 
// https://stackoverflow.com/questions/48866088/testing-a-react-mapbox-gl-with-jsodom-and-jest
jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: () => ({})
}))
jest.mock('../../tools/weatherImages')

// shallow test components
test('should render App correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<App />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render CurrentConditions correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<CurrentConditions weather={{ icon: 'fas fa-sun' }} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render Daily correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Daily />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render Hourly correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Hourly />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render Intro correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Intro />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render Weather correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Weather weatherData={mockWeatherObj}/>);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});


// def need more testing for fetch's and functionality

