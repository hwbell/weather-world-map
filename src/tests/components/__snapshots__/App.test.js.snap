// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<div
  style={Object {}}
>
  <div
    id="map"
    style={
      Object {
        "height": "100vh",
        "width": "100vw",
      }
    }
  />
  <div />
</div>
`;

exports[`should render CurrentConditions correctly 1`] = `
<div
  className="row"
  style={
    Object {
      "margin": "10",
      "marginLeft": "10vw",
    }
  }
>
  <div
    className="col"
  >
    <i
      className="undefined col"
      style={
        Object {
          "fontSize": "60px",
          "padding": "10px",
        }
      }
    />
  </div>
  <div
    className="col"
  >
    <p
      style={
        Object {
          "fontSize": "44px",
        }
      }
    >
      Â°F
    </p>
  </div>
  <div
    className="col"
  >
    <p
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": "calc(10px + 1vw)",
          "marginHorizontal": "20px",
        }
      }
    />
    <p
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": "calc(10px + 1vw)",
          "marginHorizontal": "20px",
        }
      }
    >
      undefined% humidity
    </p>
  </div>
  <div
    className="col"
  >
    <p
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": "calc(10px + 1vw)",
          "marginHorizontal": "20px",
        }
      }
    >
      undefined
    </p>
    <p
      style={
        Object {
          "color": "whitesmoke",
          "fontSize": "calc(10px + 1vw)",
          "marginHorizontal": "20px",
        }
      }
    >
      undefined
    </p>
  </div>
</div>
`;

exports[`should render Daily correctly 1`] = `
<ForwardRef
  className=""
  style={Object {}}
>
  <div
    className="row"
  >
    <HorizontalScrollContainer
      invert={false}
      onScroll={[Function]}
      sensibility={100}
      style={
        Object {
          "margin": "0px",
          "padding": "0px",
        }
      }
    />
  </div>
</ForwardRef>
`;

exports[`should render Hourly correctly 1`] = `
<ForwardRef
  className=""
  style={
    Object {
      "margin": "15px",
      "padding": "0px",
    }
  }
>
  <div
    className="row"
  >
    <HorizontalScrollContainer
      invert={false}
      onScroll={[Function]}
      sensibility={100}
    />
  </div>
</ForwardRef>
`;

exports[`should render Intro correctly 1`] = `
<ForwardRef
  pose="hidden"
  style={
    Object {
      "left": "20px",
      "position": "absolute",
      "top": "30px",
      "zIndex": 1,
    }
  }
>
  <ForwardRef
    style={
      Object {
        "color": "#283747",
        "fontSize": "calc(16px + 2vw)",
        "marginLeft": "20px",
      }
    }
  >
    take a look around
  </ForwardRef>
  <ForwardRef
    style={
      Object {
        "color": "#283747",
        "fontSize": "calc(10px + 1vw)",
        "fontWeight": "bold",
        "margin": "25px",
      }
    }
  >
    click anywhere you'd like to see the weather
  </ForwardRef>
</ForwardRef>
`;

exports[`should render Weather correctly 1`] = `
<ForwardRef
  pose="hidden"
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.75)",
      "bottom": "0px",
      "height": "550px",
      "position": "absolute",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <Button
    className="float-right"
    color="link"
    tag="button"
  >
    <i
      className="fas fa-times-circle"
      style={
        Object {
          "fontSize": "24px",
        }
      }
    />
  </Button>
  <p
    style={
      Object {
        "color": "whitesmoke",
        "fontSize": "calc(14px + 1vw)",
        "fontWeight": "bold",
        "marginLeft": "5vw",
        "padding": "10px",
      }
    }
  />
  <CurrentConditions
    weather={
      Object {
        "date": "undefined NaN, NaN",
        "humidity": undefined,
        "icon": undefined,
        "summary": undefined,
        "temperature": NaN,
        "time": "NaN:00  am",
        "uvIndex": undefined,
        "windSpeed": undefined,
      }
    }
  />
  <p
    style={
      Object {
        "color": "whitesmoke",
        "fontSize": "calc(10px + 1vw)",
        "margin": "3vh",
        "marginLeft": "10vw",
      }
    }
  />
  <Button
    color="link"
    onClick={[Function]}
    style={
      Object {
        "fontSize": "20px",
        "fontWeight": "bold",
        "marginLeft": "9vw",
        "textDecoration": "none",
      }
    }
    tag="button"
  >
    switch to 
    hourly
  </Button>
  <PoseGroup
    flipMove={true}
  >
    <ForwardRef />
    <ForwardRef>
      <Hourly
        weatherList={Array []}
      />
    </ForwardRef>
  </PoseGroup>
</ForwardRef>
`;
